src/item.cpp

```diff
@@ -535,7 +535,8 @@ long item::charges_per_volume( const units::volume &vol ) const
     if( type->volume == 0 ) {
         return INFINITE_CHARGES; // TODO: items should not have 0 volume at all!
     }
-    return count_by_charges() ? vol * type->stack_size / type->volume : vol / volume();
+    // Type cast for prevention of integer overflow with large volume containers like the cargo dimension
+    return count_by_charges() ? static_cast<int64_t>(units::to_milliliter(vol)) * static_cast<int64_t>(type->stack_size) / static_cast<int64_t>(units::to_milliliter(type->volume)) : vol / volume();
```

Comment:
More complicated than it needs to be.  You can just cast `type->stack_size` and the rest will work itself out. No need to convert or cast the volumes.